File: Zamba/ZambaWeb2013/zamba2010/Zamba.WFActivity.Regular/Action/DOInternalMessage.vb
Imports System.Collections.Generic
Imports Zamba.Core
Imports Zamba.Controls
Imports System.Xml.Serialization

<RuleCategory("Mensajes"), RuleDescription("Mensaje Interno"), RuleHelp("Permite enviar un mail interno"), RuleFeatures(False)> <Serializable()> _
Public Class DOInternalMessage
    Inherits WFRuleParent
    Implements IDOInternalMessage
    Private _isLoaded As Boolean
    Private _isFull As Boolean
    Public Overrides ReadOnly Property IsFull() As Boolean
        Get
            Return _isFull
        End Get
    End Property
    Public Overrides ReadOnly Property IsLoaded() As Boolean
        Get
            Return _isLoaded
        End Get
    End Property
    Public Overrides Sub FullLoad()

    End Sub
    Public Overrides Sub Load()

    End Sub

#Region "Atributos"
    Private _msg As InternalMessage
    Private _ToStr As String
    Private _CCStr As String
    Private _CCOStr As String
    Private _oneDocPerMail As Boolean
#End Region

#Region "Propiedades"

    Public Property Msg() As IInternalMessage Implements IDOInternalMessage.Msg
        Get
            Return _msg
        End Get
        Set(ByVal value As IInternalMessage)
            _msg = value
        End Set
    End Property

    Public Property OneDocPerMail() As Boolean Implements IDOInternalMessage.OneDocPerMail
        Get
            Return _oneDocPerMail
        End Get
        Set(ByVal value As Boolean)
            _oneDocPerMail = value
        End Set
    End Property
    Public Property CCOStr() As String Implements Core.IDOInternalMessage.CCOStr
        Get
            Return Me._CCOStr

        End Get
        Set(ByVal value As String)
            Me._CCOStr = value
        End Set
    End Property

    Public Property CCStr() As String Implements Core.IDOInternalMessage.CCStr
        Get
            Return Me._CCStr
        End Get
        Set(ByVal value As String)
            Me._CCStr = value
        End Set
    End Property

    Public Property ToStr() As String Implements Core.IDOInternalMessage.ToStr
        Get
            Return Me._ToStr
        End Get
        Set(ByVal value As String)
            Me._ToStr = value
        End Set
    End Property

#End Region

    Public Overrides Sub Dispose()

    End Sub

    Public Sub New(ByVal Id As Int64, ByVal Name As String, ByVal wfstepid As Int64, ByVal _ToStr As String, ByVal _CCStr As String, ByVal _CCOStr As String, ByVal subject As String, ByVal body As String, ByVal oneDocPerMail As Boolean)
        MyBase.New(Id, Name, wfstepid)
        Msg = New InternalMessage()
        Try
            Me.ToStr = _ToStr
            Me.CCStr = _CCStr
            Me.CCOStr = _CCOStr
            Msg.Subject = subject
            Msg.Body = body
            Me.OneDocPerMail = oneDocPerMail

        Catch ex As Exception
            Zamba.Core.ZClass.raiseerror(ex)
        End Try
    End Sub



#Region "Métodos"
    'Public Overloads Overrides Function Play(ByVal Results As SortedList) As System.Collections.SortedList
    '[Ezequiel] 09/06/2009 - Se elimino el Try-Catch
    Public Overloads Overrides Function Play(ByVal results As System.Collections.Generic.List(Of ITaskResult)) As System.Collections.Generic.List(Of ITaskResult)
        Dim playRule As New Zamba.WFExecution.PlayDOInternalMessage()
        Return playRule.Play(results, Me)
    End Function
    Public Overloads Overrides Function Playweb(ByVal results As System.Collections.Generic.List(Of ITaskResult), ByRef RulePendingEvent As RulePendingEvents, ByRef ExecutionResult As RuleExecutionResult, ByRef Params As Hashtable) As System.Collections.Generic.List(Of ITaskResult)
        Dim playRule As New Zamba.WFExecution.PlayDOInternalMessage()
        Return playRule.Play(results, Me)
    End Function
#End Region


    'Comentado por: Alejandro 09/08/07
    '
    'Private Sub BuildTo(ByVal toStr As String)
    '    toStr = "793;43;98"
    '    If toStr.Length > 0 Then

    '        Dim ids() As String = toStr.Split(";")
    '        For Each str As String In ids

    '            Dim user as iuser = UserBusiness.GetUserById(Int32.Parse(str))

    '            Dim destinatario As New Zamba.Controls.Destinatario(user, MessageType.MailTo)

    '            Msg.Destinatarios.Add(destinatario)


    '            '    Dim user as iuser
    '            '    Try
    '            '        user = UserBusiness.GetUserById(CInt(str))
    '            '    Catch ex As Exception 'Si no encuentra un usuario con este  id tira error =/
    '            '        Throw New Exception("No se pudo encontrar un usuario con id " & str)
    '            '        Exit For
    '            '    End Try
    '            '    If IsNothing(user) Then
    '            '        Throw New Exception("No se pudo encontrar un usuario con id " & str)
    '            '    Else
    '            '        If (s = String.Empty) Then
    '            '            s = user.Name
    '            '        Else
    '            '            Msg.TOUSER.Add(user)
    '            '            s = s & ";" & user.Name
    '            '        End If
    '            '    End If
    '            '    user = Nothing
    '            'Next
    '            'Msg.ToStr = toStr
    '            'toStr = Msg.ToStr+
    '        Next
    '    End If
    'End Sub
    'Private Sub BuildCC(ByVal cC As String)
    '    If cC.Length > 0 Then
    '        Dim ids() As String = cC.Split(";")

    '        For Each str As String In ids
    '            Dim user as iuser
    '            Try
    '                user = UserBusiness.GetUserById(CInt(str))
    '            Catch ex As Exception 'Si no encuentra un usuario con este  id tira error =/
    '                Throw New Exception("No se pudo encontrar un usuario con id " & str)
    '                Exit For
    '            End Try
    '            If IsNothing(user) Then
    '                Throw New Exception("No se pudo encontrar un usuario con id " & str)
    '            Else
    '                Msg.CC.Add(user)
    '                Msg.CCStr &= user.Name & " "
    '            End If
    '            user = Nothing
    '        Next
    '        Msg.CCStr = cC
    '    End If
    'End Sub
    'Private Sub BuildCCO(ByVal cCO As String)
    '    If cCO.Length > 0 Then
    '        Dim ids() As String = cCO.Split(";")
    '        For Each str As String In ids
    '            Dim user as iuser
    '            Try
    '                user = UserBusiness.GetUserById(CInt(str))
    '            Catch ex As Exception 'Si no encuentra un usuario con este  id tira error =/
    '               zamba.core.zclass.raiseerror(New Exception("No se pudo encontrar un usuario con id " & str))
    '                Exit For
    '            End Try

    '            If IsNothing(user) Then
    '               zamba.core.zclass.raiseerror(New Exception("No se pudo encontrar un usuario con id " & str))
    '            Else
    '                Msg.CCO.Add(user)
    '                Msg.CCOStr &= user.Name & " "
    '            End If
    '            user = Nothing
    '        Next
    '    End If
    'End Sub


End Class

File: Zamba/ZambaWeb2013/zamba2010/Zamba.WorkFlow.SQLExecution/Zamba.WorkFlow.Execution.ActivityLibrary/Activity/DOInternalMessage.cs
using System;
using System.ComponentModel;
using System.ComponentModel.Design;
using System.Collections;
using System.Drawing;
using System.Windows.Forms;
using System.Workflow.ComponentModel;
using System.Workflow.ComponentModel.Design;
using System.Workflow.ComponentModel.Compiler;
using System.Workflow.ComponentModel.Serialization;
using System.Workflow.Runtime;
using System.Workflow.Activities;
using System.Workflow.Activities.Rules;
using System.Reflection;
using System.Diagnostics;
using Zamba.Core;
//using Zamba.WFUserControl;
//using Zamba.Controls;

namespace Zamba.WFActivity.Xoml
{
    /// <summary>
    /// This activity show a ArrayList of delays
    /// </summary>
    [RuleDescription("MENSAJE INTERNO"), RuleHelp("Envia un mensaje interno.")]
    public partial class DOInternalMessage : ZRule, IResultActivity, IDOInternalMessage
	{
        private String _CCOStr = String.Empty, _CCStr = String.Empty, _ToStr = String.Empty;
        private Boolean _oneDocPerMail;
        private InternalMessage _msg = new InternalMessage();
        
        public DOInternalMessage()
		{
			InitializeComponent();
		}
        
        protected override ActivityExecutionStatus Execute(ActivityExecutionContext executionContext)
        {
            Zamba.WFExecution.PlayDOInternalMessage play =
                new Zamba.WFExecution.PlayDOInternalMessage();
            play.Play(Results, this);
            return ActivityExecutionStatus.Closed;
        }

        #region IDOInternalMessage Members

        string IDOInternalMessage.CCOStr
        {
            get
            {
                return this._CCOStr;
            }
            set
            {
                this._CCOStr = value;
            }
        }

        string IDOInternalMessage.CCStr
        {
            get
            {
                return this._CCStr;
            }
            set
            {
                this._CCStr = value;
            }
        }

        public IInternalMessage Msg
        {
            get
            {
                return this._msg;
            }
            set
            {
                this._msg = (InternalMessage)value;
            }
        }

        bool IDOInternalMessage.OneDocPerMail
        {
            get
            {
                return this._oneDocPerMail;
            }
            set
            {
                this._oneDocPerMail = value;
            }
        }

        string IDOInternalMessage.ToStr
        {
            get
            {
                return this._ToStr;
            }
            set
            {
                this._ToStr = value;
            }
        }

        #endregion

        public override void SetParams(object[] Params)
        {
            this._msg = (InternalMessage)Params[0];
            this._oneDocPerMail = (bool)Params[1];
        }
        public override System.Collections.Generic.List<object> GetParams()
        {
            System.Collections.Generic.List<object> Params = new System.Collections.Generic.List<object>();
            Params.Add(this._msg);
            Params.Add(this._oneDocPerMail);
            return Params;
        }

        private void BuildTo(string toStr)
        {
           if (toStr.Length > 0)
           {
               String[] ids = toStr.Split(char.Parse(";"));
               foreach (String str in ids)
               {
                   User user;
                   try
                   {
                       user = (User)UserBusiness.GetUserById(Int32.Parse(str));
                   }
                   catch 
                    {
                        throw new Exception("No se pudo encontrar un usuario con id " + str);
                    }
                    if (user == null)
                    {
                        throw new Exception("No se pudo encontrar un usuario con id " + str);
                    }
                    else
                    {
                        _msg.TOUSER.Add(user);
                        _msg.ToStr += user.Name + " ";
                    }
                    user = null;
                }
            }
        }
        private void BuildCC(string cC)
        { 
            if (cC.Length > 0)
            {
                String[] ids = cC.Split(char.Parse(";"));
                foreach (String str in ids)
                {
                    User user;
                    try
                    {
                        user = (User)UserBusiness.GetUserById(Int32.Parse(str));
                    }
                    catch
                    {
                        throw new Exception ("No se pudo encontrar un usuario con id " + str);
                    }
                    if (user == null)
                    {
                        throw new Exception("No se pudo encontrar un usuario con id " + str);
                    }
                    else
                    {
                        _msg.CC.Add(user);
                        _msg.CCStr += user.Name + " ";
                    
                    }
                    
                user = null;
                }
                _msg.CCStr = cC;
            }
        }
        private void BuildCCO(string cCO)
        { 
            if (cCO.Length > 0)
            {
                String[] ids = cCO.Split(char.Parse(";"));
                foreach (String str in ids)
                {
                    User user;
                    try
                    {
                        user = (User)UserBusiness.GetUserById(Int32.Parse(str));
                    }
                   catch 
                    {
                       ZClass.raiseerror(new Exception("No se pudo encontrar un usuario con id " + str));
                       break;
                    }
                    if (user == null)
                    { 
                        ZClass.raiseerror(new Exception("No se pudo encontrar un usuario con id " + str));
                    }
                    else
                    {
                        _msg.CCO.Add(user);
                        _msg.CCOStr += user.Name + " ";
                    }
                    user = null;
                }
            }
        }
    }
}